pkg/ck8s/workload_cluster.go

```diff
@@ -259,17 +285,76 @@ func (w *Workload) RefreshCertificates(ctx context.Context, machine *clusterv1.M
 		return 0, fmt.Errorf("failed to refresh certificates: %w", err)
 	}
 
+	return planResponse.Seed, nil
+}
+
+func (w *Workload) refreshCertificatesRun(ctx context.Context, machine *clusterv1.Machine, nodeToken string, request *apiv1.ClusterAPICertificatesRunRequest) (int, error) {
+	runResponse := &apiv1.ClusterAPICertificatesRunResponse{}
+	header := map[string][]string{
+		"node-token": {nodeToken},
+	}
+
+	k8sdProxy, err := w.GetK8sdProxyForMachine(ctx, machine)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create k8sd proxy: %w", err)
+	}
+
+	if err := w.doK8sdRequest(ctx, k8sdProxy, http.MethodPost, "1.0/x/capi/refresh-certs/run", header, request, runResponse); err != nil {
```

Comment:
Maybe we need to replace this with: https://github.com/canonical/k8s-snap-api/blob/main/api/v1/rpc_clusterapi_certificates_run.go#L3

Reply:
Unfortunately, we cannot replace the path here with the k8s-snap-api const, because it's missing the version prefix. I wonder if using something like string concatenation would be optimized by the compiler.

Reply:
It's seems that the compiler optimizes this behavior. I'll create a follow up PR to address this.